core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -322,6 +325,7 @@ public void testAcknowledgementCommitCallbackSuccessfulAcknowledgement(String pe
         ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(tp.topic(), tp.partition(), null, "key".getBytes(), "value".getBytes());
         KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer());
```

Comment:
should the closeable resources be closed in a try / finally block so that the resources are closed even if we have an assertion failure?

Reply:
Sure, that's probably a good thing to do. I'll put in the next commit.

Reply:
Done, with the exception of the tests which use ExecutorService. I'll see if I can improve those too.